### ISSUE TITLE
AWS-Neptune: Add replica for multi region on AWS Neptune

### ISSUE THREAD

**Issue by hugo-alvarez-bold** [aws/aws-cdk#27988]
### Describe the feature

We need a feature on CDK to add replica data on Neptune for my services that use multiple regions

### Use Case

We need to implement a multi region services in our architecture for regulations purpose on our fintech company, and we use this service to avoid fraud in our transactions.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.83.1

### Environment details (OS name and version, etc.)

python 3.7.6

**Comment by pahud** [comment]
Makes sense. At this moment, [aws-neptune](https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-neptune) does not have L2 constructs. If cloudformation has native support that, we can build a L2 with the support, otherwise the custom resources might be required.

**Comment by hugo-alvarez-bold** [comment]
looking in the docs on Cloudformation Neptune, there is not an attribute or class like dynamoDB GlobalTable can help with this either

**Comment by triggan** [comment]
Amazon Neptune has a feature called [Neptune Global Database](https://docs.aws.amazon.com/neptune/latest/userguide/neptune-global-database.html) which allows you to have read-only secondary clusters in separate regions from the primary cluster.  At present, this feature does not have CloudFormation support.  You would need to use a CustomResource construct to implement this in CDK.

**Comment by masterhugo** [comment]
Hi @pahud and @triggan!

Just a quick update on this thread: since the last comments in late 2023, AWS CloudFormation has officially added native support for AWS::Neptune::GlobalCluster.

Taking advantage of this, I've just opened PR #38221 implementing the L2 constructs for it, without needing to rely on Custom Resources! 

I would love to get your thoughts and review on the PR whenever you have some time. Thanks!
